commit f807f2c07f1cef301088b808363ea4ee81a80f84
Author: Wez Furlong <wez@fb.com>
Date:   Thu Jan 24 20:35:08 2019 -0800

    watchman: fixup cmake install

    Summary:
    * Fixed setting up the watchman state dir to be based on the cmake install prefix
    @nocommit undone: * Fixed python installation to respect DESTDIR

    Reviewed By: simpkins

    Differential Revision: D13747960

    fbshipit-source-id: b2357d94343ccadc0127400d3d8b513088f7763c

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f2bab746..2e20885e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,13 +26,19 @@ endfunction()

 file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/config.h.new" "#pragma once\n")

+if(NOT CMAKE_SYSTEM_NAME STREQUAL "Windows")
+  set(WATCHMAN_STATE_DIR "${CMAKE_INSTALL_PREFIX}/var/run/watchman")
+  install(DIRECTORY DESTINATION ${WATCHMAN_STATE_DIR} DIRECTORY_PERMISSIONS
+    OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE
+    GROUP_EXECUTE WORLD_READ WORLD_WRITE WORLD_EXECUTE SETGID)
+endif()
+
 config_h("// Generated by cmake")
 if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
   config_h("#define WATCHMAN_CONFIG_FILE \
 \"C:/ProgramData/facebook/watchman.json\"")
 else()
   config_h("#define WATCHMAN_CONFIG_FILE \"/etc/watchman.json\"")
-  set(WATCHMAN_STATE_DIR "/var/run/watchman")
   config_h("#define WATCHMAN_STATE_DIR \"${WATCHMAN_STATE_DIR}\"")
 endif()
 config_h("#define PACKAGE_VERSION \"${PACKAGE_VERSION}\"")
@@ -391,13 +397,6 @@ endif()

 install(TARGETS watchman RUNTIME DESTINATION bin)

-if(NOT CMAKE_SYSTEM_NAME STREQUAL "Windows")
-  set(DEST_STATE_DIR "$ENV{DESTDIR}${WATCHMAN_STATE_DIR}")
-  install(DIRECTORY DESTINATION ${DEST_STATE_DIR} DIRECTORY_PERMISSIONS
-    OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE
-    GROUP_EXECUTE WORLD_READ WORLD_WRITE WORLD_EXECUTE SETGID)
-endif()
-
 set(tests)
 # Helper function to define a unit test executable
 function(t_test NAME)
